src/lib/common/string.cpp

```diff
@@ -1082,3 +1082,103 @@ void toLowercase(char* s)
     ++s;
   }
 }
+
+
+#ifdef PARANOID_JSON_INDENT
+// -----------------------------------------------------------------------------
+//
+// jsonFix - replace all ':' with ' :' in a JSON buffer (as some other fixes)
+//
+// 01. Count number of replacements - to get necessary size of out buffer
+// 02. Go over the in buffer and copy one by one in -> out, avoiding replacements
+//     while inside strings.
+//
+char* jsonFix(const char* in)
+{
+  int          outSize   = 1;  // Room for zero-termination char
+  const char*  inP       = in;
+  char*        out;
+  bool         inString  = false;
+
+  while (*inP != 0)
+  {
+    // If backslash, step over it AND the next char
+    if (*inP == '\\')
+    {
+      outSize += 2;
+      inP     += 2;
+      continue;
+    }
+
+    // If double-quote, change state (inString false/true)
+    if (*inP == '"')
+    {
+      inString = (inString == true)? false : true;
+      ++outSize;
+      ++inP;
+      continue;
+    }
+
+    // If not inside a string, detect ':' and add an extra byte if found
+    if (!inString)
+    {
+      if (*inP == ':')
+      {
+        ++outSize;  // make room for an extra space
+      }
+    }
+
+    // Step over current char and continue
+    ++outSize;
+    ++inP;
+  }
+  //printf("in-size: %lu, out-size: %d\n", strlen(in), outSize);
```

Comment:
This seems like an old line of debug that should be removed

Reply:
This is within a #ifdef PARANOID_JSON_INDENT block, so it will be removed soon.

Reply:
No probs

NTC